File: nodejs/number-of-lines-to-write-string806.js
'use strict';
/**
 * @param {number[]} widths
 * @param {string} S
 * @return {number[]}
 */
var numberOfLines = function(widths, S) {
    let count = 0;
    let sum = 0;

    for (let i = 0; i < S.length; i++) {
        let val = widths[S.charCodeAt(i)-97];
        if (sum+val>100) {
            count++;
            sum = 0;
        }

        sum += val;
    }
    return [count+1,sum];
};


let widths = [4,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10];
let S = "bbbcccdddaaa";
console.log(numberOfLines(widths,S));

File: golang/numberoflinestowritestring806/number-of-lines-to-write-string806.go
package main

import "fmt"

func main() {
	widths := []int{10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10}
	S := "abcdefghijklmnopqrstuvwxyz"
	fmt.Println(numberOfLines(widths[0:len(widths)], S))
}

func numberOfLines(widths []int, S string) []int {
	sum := 0
	count := 0

	for _, v := range S {
		val := widths[int(v)-97]
		if val+sum > 100 {
			count++
			sum = 0
		}
		sum += val
		// fmt.Println(v)
	}
	return []int{count + 1, sum}
}
